socs/db/suprsync.py

```diff
@@ -208,6 +212,70 @@ def __init__(self, db_path, create_all=True, echo=False):
         if create_all:
             Base.metadata.create_all(self._engine)
 
+    def get_archive_stats(self, session=None):
+        if session is None:
+            session = self.Session()
+
+        archive_names = session.query(SupRsyncFile.archive_name)\
+            .distinct().all()
+
+        stats = {}
+        for name, in archive_names:
+            files = session.query(SupRsyncFile).filter(
+                SupRsyncFile.archive_name == name,
+            ).order_by(asc(SupRsyncFile.timestamp)).all()
+
+            finalized_until = None
+            num_files_to_copy = 0
+            last_file_added = ''
+            last_file_copied = ''
+
+            for f in files:
+                last_file_added = f.local_path
+                if (f.local_md5sum == f.remote_md5sum):
+                    last_file_copied = f.local_path
+
+                if (not f.ignore) and (f.local_md5sum != f.remote_md5sum):
+                    num_files_to_copy += 1
+
+                if finalized_until is None and not (f.ignore):
+                    if f.local_md5sum != f.remote_md5sum:
+                        finalized_until = f.timestamp - 1
```

Comment:
Same question here, are file guaranteed to be spaced out enough that `timestamp - 1` is a good cut off for "finalized until" times?